### ISSUE TITLE
Checkra1n crash after enabling A11 BR Check 

### ISSUE THREAD

**Issue by elouannd** [raspberryenvoie/odysseyn1x#23]
When I have my IPhone plugged and I enable the skip A11 Br check, it’s freeze instantly. But when unplug my iPhone and I enable the skip A11 everything’s is fine. And when I plug it again it’s showing Lockdown error -19 and freeze just after 
I’m on 2.10

**Comment by raspberryenvoie** [comment]
You need to remove the password from your iPhone. Did you do it? From checkra.in:
> A11 devices on iOS 14.0 and above require removing the passcode and enabling “Skip A11 BPR check” in the options. This is not recommended, but it’s there.

Also which iPhone are you trying to jailbreak?

**Comment by chrisi0801** [comment]
Have the same issue with my iPhone X.
When selecting "Skip A11 BPR check", the script freezes
Passcode is of course removed

btw, thank you for updating the script so fast :)

**Comment by elouannd** [comment]
I'm using an IPhone X and yes I've already removed the password and reset Face ID

**Comment by raspberryenvoie** [comment]
Does this problem still occur with version 2.11?

**Comment by chrisi0801** [comment]
I will try it again now with 2.11 on ios 14.5

**Comment by chrisi0801** [comment]
I tried it again. It doesn't work. As soon as you enable the BPR check, the script freezes. Strange thing: When enabling "Allow untested versions", the script still shows "Sorry, Iphone X... on Ios 14.5 is not supported". Although the new checkra1n version should be compatible with ios 14.5 out of the box

**Comment by raspberryenvoie** [comment]
Wait what freezes?

**Comment by chrisi0801** [comment]
The whole odysseyn1x script. You have to reset the pc to get it working again

**Comment by raspberryenvoie** [comment]
Do you mean the os?

**Comment by chrisi0801** [comment]
Yes exactly. The OS crashes. And also very strange is the fact that it shows that my device isn't compatible although it is definitely compatible

**Comment by chrisi0801** [comment]
I am trying to prepare a video for you. I will send it to you as soon as it's done

**Comment by chrisi0801** [comment]
I sent you a YouTube link via e-mail

**Comment by raspberryenvoie** [comment]
Thank you!

**Comment by raspberryenvoie** [comment]
I've just tried that in odysseyn1x in VirtualBox and I get the same problem

**Comment by raspberryenvoie** [comment]
I've just tried that on odysseyn1x v2.11 in VirtualBox and the OS doesn't freeze.

@chrisi0801 and @mcqwerty1907:
Does the OS only freeze with checkra1n version 0.12.3?
Does this issue only occur on odysseyn1x (or any Linux distro)?
Can you tell me what your hardware is?

**Comment by chrisi0801** [comment]
@raspberryenvoie Can you share your VirtualBox settings with me, so I can also try it there?
The OS only freezes with checkra1n 0.12.3
I will test other linux distros tomorrow
Hardware: i7 8700k, 1080ti, Asus Prime Z370-A

**Comment by raspberryenvoie** [comment]
Create a 64-bit Debian VM with default settings. Go into the VM settings > Storage. After that, click on the the CD button next to "Controller: IDE". Add the 64-bit odysseyn1x ISO. Save the settings then boot the VM.

Note that you won't be able to jailbreak an iPhone inside the VM.

**Comment by chrisi0801** [comment]
Okay so it's definitely not hardware related. I tried with my laptop too and it doesn't work either. It's related to either the Iphone X or the OS. I will try 2.10 now

**Comment by chrisi0801** [comment]
Okay no surprise, 2.10 doens't work either. The last version that actually worked for me was 2.8 
With 2.9 I had an issue with BPR check too, but not a freezing issue, I had a warning that the passcode is not disabled. it could also be checkra1n itself that is completely broken with the iphone x

**Comment by raspberryenvoie** [comment]
Okay. Tell me if it only freezes if your iPhone X is plugged to the PC because I don't have an iPhone X and I'd find out what's going on.

**Comment by elouannd** [comment]
It's don't freezing when my iPhone is not plugged put when I'm plugging it it's freeze

**Comment by chrisi0801** [comment]
I will install ubuntu natively now on my laptop and I will try checkra1n there and hope it works

**Comment by chrisi0801** [comment]
checkra1n worked fine on native ubuntu. First I had USB issues but i remembered that you have to turn on secure boot.
The odysseyra1n script works perfectly fine in odysseyn1x 2.11 so no issues there. The issue lies in checkra1n. Hope you can find out whats actually wrong

**Comment by raspberryenvoie** [comment]
You used checkra1n cli in the terminal, not the GUI version, right?

Thank you so much for investigating!

**Comment by PWittix** [comment]
I experienced the same issue with my iPhone X iOS 14.5 with odysseyn1x 2.11.  I was able to get checkra1n working with the following steps:

1. Start with powered off computer
2. Connect device
3. Power on, then open checkra1n.  Should say unsupported a11 message.
4. Change checkra1n options: A11 skip bpr check
5. Should get error -19.  Click trust on the device. 
6. Open options, but don't change any options.
7. Should be able to use checkra1n.

If the checkra1n gui freezes, press Ctrl + C to exit the program and bring up a console terminal, then type exit.  This should return to the original menu screen.

**Comment by chrisi0801** [comment]
I can confirm that @PWittix workaround works.
@raspberryenvoie I tested both CLI and GUI and both worked. But I prefer the CLI variant.

**Comment by raspberryenvoie** [comment]
According to @PWittix just checkra1n freezes not the whole OS. I found another possible workaround on the checkra1n bugtracker. Could you try it? Run checkra1n normally but before plugging your iPhone X make sure it's unlocked.

**Comment by chrisi0801** [comment]
I didn't know that you can exit checkra1n inside the OS, so not the OS is crashing, it's just checkra1n. I will test your suggestion now.

**Comment by chrisi0801** [comment]
I tried it now @raspberryenvoie and this doesn't work for me. The only working fix is the above by @PWittix. And yes I confirm that only checkra1n crashes, not the OS. Basically, as soon as you enable BPR check checkra1n freezes. It doesn't matter which way round: First BPR check, then plugging in the phone -> Freeze , First plugging in and then BPR check -> Freeze. And it also doesn't matter if the phone is unlocked or not and if the phone trusts the device. Only @PWittix fix works

**Comment by raspberryenvoie** [comment]
Weird. It seems like it's a checkra1n issue. If so it should be reported to the checkra1n bug tracker. However I'm not sure since you said checkra1n works on Ubuntu

**Comment by aliefe04** [comment]
its a checkra1n issue but you can use cli

**Comment by dmatora** [comment]
is there a way to skip A11 BPR check in cli mode?

**Comment by LaxusiC** [comment]
Is there a solution to fix this issue?

**Comment by raspberryenvoie** [comment]
> Is there a solution to fix this issue?

Have you tried @PWittix 's workaround?

**Comment by Verwelkt** [comment]
Just wanted to say that I've also had this issue. iPhone 8 Plus running iOS 14.7.1. However, instead of occurring while A11 bypass is enabled, it seems to be due to allowing unsupported versions of iOS.

**Comment by raspberryenvoie** [comment]
> Just wanted to say that I've also had this issue. iPhone 8 Plus running iOS 14.7.1. However, instead of occurring while A11 bypass is enabled, it seems to be due to allowing unsupported versions of iOS.

Does the above workaround work? Other than that I can't help you because it's an issue related to checkra1n

**Comment by BinhPQ2** [comment]
> I experienced the same issue with my iPhone X iOS 14.5 with odysseyn1x 2.11. I was able to get checkra1n working with the following steps:
> 
> 1. Start with powered off computer
> 2. Connect device
> 3. Power on, then open checkra1n.  Should say unsupported a11 message.
> 4. Change checkra1n options: A11 skip bpr check
> 5. Should get error -19.  Click trust on the device.
> 6. Open options, but don't change any options.
> 7. Should be able to use checkra1n.
> 
> If the checkra1n gui freezes, press Ctrl + C to exit the program and bring up a console terminal, then type exit. This should return to the original menu screen.

Holy shit it works

**Comment by cs59** [comment]
> > Just wanted to say that I've also had this issue. iPhone 8 Plus running iOS 14.7.1. However, instead of occurring while A11 bypass is enabled, it seems to be due to allowing unsupported versions of iOS.
> 
> Does the above workaround work? Other than that I can't help you because it's an issue related to checkra1n

Hey, I also have the same problem on 8+ 14.4.1, what happened is that i have downgraded my iphone from 14.8 to 14.4.1 using future restore using 15.4 sep, but after downgraded when i tried to to enable skip A11 bpr check and when i click back then checkra1n get freezes if my device is plugged in, and if not then it will normal but as soon as I connect my device then it get freezed again (skip A1- check already ticked) so that’s only happening becoz of skip A11 check also my laptop gets extremely heated don’t know why, 
So now, is there any way i can do jb? Am on hello screen currently because i have to bypass the device also the solution did not worked by @pt…. Whatever his name,
Pls say what to do now?

**Comment by raspberryenvoie** [comment]
Sorry I can't help you with this as I don't own a device that requires skipping A11 BPR check.
